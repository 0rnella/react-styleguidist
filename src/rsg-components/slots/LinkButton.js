import React from 'react';
import PropTypes from 'prop-types';
import MdLink from 'react-icons/lib/md/link';
import ToolbarButton from 'rsg-components/ToolbarButton';
import { getUrl } from '../../utils/utils';

const LinkButton = ({ slug }) => (
	<ToolbarButton href={getUrl({ slug, anchor: true })} title="Link to this component">
		<MdLink />
	</ToolbarButton>
);

LinkButton.propTypes = {
	slug: PropTypes.string.isRequired,
};

export default LinkButton;
