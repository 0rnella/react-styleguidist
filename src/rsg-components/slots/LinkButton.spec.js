import React from 'react';
import LinkButton from './LinkButton';

it('should render an anchor link', () => {
	const actual = shallow(<LinkButton slug="Pizza" />);

	expect(actual).toMatchSnapshot();
});
