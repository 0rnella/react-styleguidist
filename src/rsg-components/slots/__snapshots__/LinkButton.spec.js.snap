// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an anchor link 1`] = `
<Styled(ToolbarButton)
  href="blank#Pizza"
  title="Link to this component"
>
  <MdLink />
</Styled(ToolbarButton)>
`;
